test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/PartialTagHelperTest.cs

```diff
@@ -647,7 +647,193 @@ public void ResolveModel_ThrowsWhenNullModelAndForProvided()
             var content = HtmlContentUtilities.HtmlContentToString(output.Content, new HtmlTestEncoder());
             Assert.Empty(content);
         }
+        
+        [Fact]
+        public async Task ProcessAsync_RendersMainPartial_If_FallbackIsSet_AndMainPartialIsFound()
+        {
+            // Arrange
+            var expected = "Hello from partial!";
+            var bufferScope = new TestViewBufferScope();
+            var partialName = "_Partial";
+            var fallbackName = "_Fallback";
+            var model = new object();
+            var viewContext = GetViewContext();
+
+            var view = new Mock<IView>();
+            view.Setup(v => v.RenderAsync(It.IsAny<ViewContext>()))
+                .Callback((ViewContext v) =>
+                {
+                    v.Writer.Write(expected);
+                })
+                .Returns(Task.CompletedTask);
+
+            var fallbackView = new Mock<IView>();
+            fallbackView.Setup(v => v.RenderAsync(It.IsAny<ViewContext>()))
+                .Callback((ViewContext v) =>
+                {
+                    v.Writer.Write("Hello from fallback partial!");
+                })
+                .Returns(Task.CompletedTask);
+
+            var viewEngine = new Mock<ICompositeViewEngine>();
+            viewEngine.Setup(v => v.GetView(It.IsAny<string>(), partialName, false))
+                .Returns(ViewEngineResult.Found(partialName, view.Object));
+            viewEngine.Setup(v => v.GetView(It.IsAny<string>(), fallbackName, false))
+                .Returns(ViewEngineResult.Found(fallbackName, fallbackView.Object));
+
+            var tagHelper = new PartialTagHelper(viewEngine.Object, bufferScope)
+            {
+                Name = partialName,
+                ViewContext = viewContext,
+                Fallback = fallbackName
+            };
+            var tagHelperContext = GetTagHelperContext();
+            var output = GetTagHelperOutput();
+
+            // Act
+            await tagHelper.ProcessAsync(tagHelperContext, output);
+
+            // Assert
+            var content = HtmlContentUtilities.HtmlContentToString(output.Content, new HtmlTestEncoder());
+            Assert.Equal(expected, content);
+        }
+
+        [Fact]
+        public async Task ProcessAsync_IfHasFallback_Throws_When_MainPartialAndFallback_AreNotFound()
+        {
+            // Arrange
+            var bufferScope = new TestViewBufferScope();
+            var partialName = "_Partial";
+            var fallbackName = "_Fallback";
+            var expected = string.Join(Environment.NewLine,
```

Comment:
Had a chat with @Eilon about the error message. The suggestion was to include both of them with some variation suggesting the second set of view locations are the fallback e.g.

```
The partial view '{partialName}` was not found. The following locations were searched:
PartialNotFound1
PartialNotFound2

The fallback view {fallbackName} was not found. The following locations were searched:
FallbackNotFound1
FallbackNotFound2
```

I can make this change as part of merging the PR